code/modules/mob/hear_say.dm

```diff
@@ -89,132 +45,95 @@
 	if(!client)
 		return
 
-	if(sleeping || stat==1) //If unconscious or sleeping
-		hear_sleep(message)
-		return
-
-	var/track = null
+	var/speaker_name = get_hear_name(speaker, hard_to_hear, vname)
 
-	//non-verbal languages are garbled if you can't see the speaker. Yes, this includes if they are inside a closet.
-	if (language && (language.flags & NONVERBAL))
-		if (!speaker || (src.sdisabilities & BLIND || src.blinded) || !(speaker in view(src)))
-			message = stars(message)
-
-	if(!(language && (language.flags & INNATE))) // skip understanding checks for INNATE languages
-		if(!say_understands(speaker,language))
-			if(isanimal(speaker))
-				var/mob/living/simple_animal/S = speaker
-				if(S.speak && S.speak.len)
-					message = pick(S.speak)
-				else
-					return
-			else
-				if(language)
-					message = language.scramble(message)
-				else
-					message = stars(message)
+	if(language)
+		message = language.format_message_radio(message, verb)
+	else
+		message = "[verb], <span class=\"body\">\"[message]\"</span>"
 
-		if(hard_to_hear)
-			message = stars(message)
+	on_hear_radio(part_a, speaker_name, part_b, message)
 
+/mob/proc/get_hear_name(var/mob/speaker, hard_to_hear, vname)
```

Comment:
"vname" doesn't look like appropriate variable name, please rename it all over the pull request.